site/blocks/StickyNav.tsx

```diff
@@ -0,0 +1,179 @@
+                    const containerPadding = 32
+                    current.scrollTo({
+                        behavior: "smooth",
+                        left: listItem.offsetLeft - containerPadding,
+                    })
+                }
+            }
+        }
+
+        this.setState({
+            currentHeadingIndex,
+        })
+    }
+
+    setHeadingPositions() {
+        const headingPositions: any[] = []
+        this.links.forEach((link) => {
+            const element = document.querySelector<HTMLElement>(link.target)
+            if (element) {
+                const { y } = getTotalOffset(element)
+                headingPositions.push({ id: link.target, position: y })
+            }
+        })
+
+        this.setState(
+            {
+                headingPositions,
+            },
+            () => {
+                this.setCurrentHeading()
+            }
+        )
+    }
+
+    componentDidMount() {
+        // Web fonts change the position of elements
+        // so we set the active heading only once we know where everything is.
+        // The alternative is to call setHeadingPositions as soon as the component mounts,
+        // but this can make currentHeadingIndex jump around.
+        window.addEventListener("load", this.handleResize)
+        window.addEventListener("scroll", this.handleScroll)
+        window.addEventListener("resize", this.handleResize)
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener("scroll", this.handleScroll)
+        window.removeEventListener("resize", this.handleResize)
+        window.removeEventListener("load", this.handleResize)
+    }
+
+    render() {
+        if (!this.links) return null
+        return (
+            <>
+                <style>
+                    {`
+                    [id] {
+                         scroll-margin-top:${
+                             STICKY_NAV_HEIGHT + HEADING_SCROLL_MARGIN
+                         }px;
+                    }`}
+                </style>
+                <ul className="sticky-nav-container" ref={this.ulRef}>
+                    {this.links.map((link, i) => (
+                        <li key={link.target}>
+                            <a
+                                tabIndex={0}
+                                className={classnames({
+                                    active:
+                                        i === this.state.currentHeadingIndex,
+                                })}
+                                href={link.target}
+                            >
+                                {link.text}
+                            </a>
+                        </li>
+                    ))}
+                </ul>
+            </>
+        )
+    }
+}
+
+export default StickyNav
+
+export const hydrateStickyNav = () => {
+    const wrapper = document.querySelector(".sticky-nav")
+    const anchorTags = document.querySelectorAll(".sticky-nav a")
```

Comment:
If you're using
```suggestion
    const anchorTags = document.querySelectorAll<HTMLElement>(".sticky-nav a")
```
, then you hopefully won't need to use lodash's `get` below to get text and target.